src/util/hitbox/index.ts

```diff
@@ -0,0 +1,199 @@
+        Math.abs(x1 - x2),
+        Math.abs(y1 - y2)
+    )
+    return box.toPolygon()
+}
+
+/**
+ * Test the selection polygon against all strokes with multiple hitboxe
+ * segments and return a set of all collided stroke IDs.
+ */
+export function matchStrokeCollision(
+    strokes: StrokeCollection,
+    selectionPolygon: Polygon
+): StrokeCollection {
+    const result: StrokeCollection = {}
+    Object.keys(strokes).forEach((id) => {
+        // test each hitbox segment
+        for (let i = 0; i < (strokes[id].hitboxes ?? []).length; i += 1) {
+            if (
+                testPolygonPolygon(
+                    (strokes[id].hitboxes ?? [])[i],
+                    selectionPolygon
+                )
+            ) {
+                result[id] = strokes[id]
+                break
+            }
+        }
+    })
+    return result
+}
+
+interface GetEllipseOutlineProps {
+    x: number //  ellipse center (x-axis)
+    y: number //  ellipse center (y-axis)
+    rx: number // ellipse radius (x-axis)
+    ry: number //  ellipse radius (y-axis)
+    // specify how many segments to divide each ellipse quarter into
+    // => 4 segmentsPerQuarter = 16 segments total
+    segmentsPerQuarter: number
+}
+
+export const getEllipseOutline = ({
+    x,
+    y,
+    rx,
+    ry,
+    segmentsPerQuarter,
+}: GetEllipseOutlineProps) => {
+    const outlinePoints = new Array(segmentsPerQuarter * 4)
+
+    // Set top, bottom, left, right points
+    outlinePoints[0] = { x: x + rx, y }
+    outlinePoints[segmentsPerQuarter] = { x, y: y + ry }
+    outlinePoints[2 * segmentsPerQuarter] = { x: x - rx, y }
+    outlinePoints[3 * segmentsPerQuarter] = { x, y: y - ry }
+
+    for (let i = 1; i < segmentsPerQuarter; i++) {
+        const angleRadian = (i * Math.PI) / (segmentsPerQuarter * 2)
+
+        const xOff =
+            (rx * ry) /
+            Math.sqrt(ry ** 2 + rx ** 2 * Math.tan(angleRadian) ** 2)
+        const yOff = Math.tan(angleRadian) * xOff
+
+        outlinePoints[i] = {
+            x: x + xOff,
+            y: y + yOff,
+        }
+        outlinePoints[2 * segmentsPerQuarter - i] = {
+            x: x - xOff,
+            y: y + yOff,
+        }
+        outlinePoints[2 * segmentsPerQuarter + i] = {
+            x: x - xOff,
+            y: y - yOff,
+        }
+        outlinePoints[4 * segmentsPerQuarter - i] = {
+            x: x + xOff,
+            y: y - yOff,
+        }
+    }
+
+    return outlinePoints
+}
+
+export const getTransformedPoints = (stroke: Stroke): Stroke["points"] =>
+    stroke.points.map((value, index) =>
+        index % 2 === 0
+            ? (value + stroke.x) * stroke.scaleX
+            : (value + stroke.y) * stroke.scaleY
+    )
+
+/**
+ * Get the hitboxes array of a stroke
+ */
+export const getStrokeHitbox = (stroke: Stroke): Polygon[] => {
+    const hitboxes: Polygon[] = []
+
+    switch (stroke.type) {
+        case ToolType.Line:
+        case ToolType.Highlighter:
+        case ToolType.Pen: {
```

Comment:
Let's keep this logic within `stroke/`. `util/hitbox` are generic util functions that should depend on stroke internals (eg.`stroke.type`)